### PR TITLE
Optionally generate prompt datasource accesses.

### PR DESCRIPTION
**Pull request by pgavlin**
Just what it says on the tin. Only works for NodeJS for the time being.

Part of https://github.com/pulumi/home/issues/543.

### CODE REVIEW

**Comment by pgavlin** [comment]
This turned out to be rather straightforward. I will add some tests presently.

File: gen/nodejs/generator.go [hunk 1]
```diff
@@ -341,30 +350,8 @@ func (g *generator) GeneratePreamble(modules []*il.Graph) error {
...
-		for _, n := range m.Locals {
-			_, err := il.VisitBoundNode(n.Value, findOptionals, il.IdentityVisitor)
-			contract.Assert(err == nil)
-		}
-		for _, n := range m.Variables {
-			_, err := il.VisitBoundNode(n.DefaultValue, findOptionals, il.IdentityVisitor)
-			contract.Assert(err == nil)
-		}
+		err := il.VisitAllProperties(m, findOptionals, il.IdentityVisitor)
+		contract.Assert(err == nil)
```
Comment (CyrusNajmabadi) [inline gen/nodejs/generator.go]: this seems like an odd kind of situation.  if we're assrting that errors can't happen, why have the Visit system able to return errors at all?  Note: this is just for discussion.  I know that this PR isn't introducing this concept.  It just feels strange to me.

**Comment by CyrusNajmabadi** [comment]
Nice!  You're the general owner here, so i leave this sort of design decision to you.  However, I find it somewhat questionable to add the option here.  It feels like we move forward rapidly enough, and can make calls like: "all our TF SDKs shoudl be on a recent enough `@pulumi/pulumi`.  As such, we just emit the synchronous version and leave it at that". 

I just like the idea of keeping the codebase simple and single-purposed, rather than trying to have it support many use cases, most of which will not be necessary in the future.

**Comment by pgavlin** [comment]
Example of the new codegen in the AWS docs: https://github.com/pulumi/pulumi-aws/compare/pgavlin/tf2pulumi

**Comment by lukehoban** [comment]
This won’t currently work will it?  We don’t (yet?) accept Inputs as Inputs on data sources - right?

https://github.com/pulumi/pulumi-aws/compare/pgavlin/tf2pulumi#diff-2a510a16e214e44c39922f9e4dfc686aR20

**Comment by pgavlin** [comment]
> This won’t currently work will it? We don’t (yet?) accept Inputs as Inputs on data sources - right?

There's no `Input` involved there: both of those calls are data sources, so `myRestApi.id` is a prompt value.

**Comment by lukehoban** [comment]
Ah - indeed (the `apply` threw me off). Diff looks good to me.

**Pull request merged**